modules/core/shared/src/main/scala/coursier/core/Reconciliation.scala

```diff
@@ -0,0 +1,118 @@
+package coursier.core
+
+/**
+ * Represents a reconciliation strategy given a dependency conflict.
+ */
+abstract class Reconciliation {
+  /**
+   * Reconcile multiple version candidate.
+   *
+   * Returns `None` in case of conflict.
+   */
+  def reconcileVersions(module: Module, versions: Seq[String]): Option[String]
```

Comment:
Rename to `apply`? (down the line, we have `reconciler.reconcileVersions(…)`, which feels heavyweight)

Reply:
Let's not pass `module` either if it's not needed.

Reply:
Yea. If we are making it sealed, then I won't pass more than needed.